File: examp-backend/src/main/java/com/dipta/eb/services/QuizService.java
package com.dipta.eb.services;

import com.dipta.eb.models.exam.Category;
import com.dipta.eb.models.exam.Quiz;
import org.springframework.http.ResponseEntity;

import java.util.List;
import java.util.Set;

public interface QuizService {
    Quiz addQuiz(Quiz quiz);
    Quiz updateQuiz(Quiz quiz);
    Set<Quiz> getQuizzes();
    Quiz getQuiz(Long qId);
    void deleteQuiz(Long qId);


    public List<Quiz> getQuizzesOfCategory(Category category);
    public List<Quiz> getActiveQuiz();
    public List<Quiz> getActiveQuizOfCategory(Category category);
}

File: examp-frontend/src/app/services/quiz.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { APIURL } from './helper';

@Injectable({
  providedIn: 'root',
})
export class QuizService {
  constructor(private _http: HttpClient) {}
  //get
  public getQuiz() {
    return this._http.get(`${APIURL}/quiz/`);
  }
  //add
  public addQuiz(quiz:any) {
    return this._http.post(`${APIURL}/quiz/`, quiz);
  }
  //delete
  public deleteQuiz(qId:any){
    return this._http.delete(`${APIURL}/quiz/${qId}`);
  }

  // single quiz
  public getSingleQuiz(qId:any){
    return this._http.get(`${APIURL}/quiz/${qId}`);
  }

  //update
  public updateQuiz(quiz:any){
return this._http.put(`${APIURL}/quiz/`,quiz)
  }

  public findQuizByCategory(cId:any){
    return this._http.get(`${APIURL}/quiz/category/${cId}`)
  }

  public getActiveQuiz(){
    return this._http.get(`${APIURL}/quiz/active/`);
  }

  public getActiveQuizOfCategory(cId:any){
    return this._http.get(`${APIURL}/quiz/category/active/${cId}`)
  }
  
}
